src/Protos/Grpc/streams.v2.proto

```diff
@@ -0,0 +1,184 @@
+message AppendStreamSuccess {
+	// The name of the stream to which records were appended.
+	string stream = 1;
+
+	// The position of the last appended record in the stream.
+	optional int64 position = 2;
+
+	// The revision of the stream after the append operation.
+	int64 stream_revision = 3;
+}
+
+// Failure represents the detailed error information when an append operation fails.
+message AppendStreamFailure {
+	// The name of the stream to which records failed to append.
+	string stream = 1;
+
+	// The error details
+	oneof error {
+		// Failed because the actual stream revision didn't match the expected revision.
+		ErrorDetails.WrongExpectedRevision wrong_expected_revision = 2;
+
+		// Failed because the client lacks sufficient permissions.
+		ErrorDetails.AccessDenied access_denied = 3;
+
+		// Failed because the target stream has been deleted.
+		ErrorDetails.StreamDeleted stream_deleted = 4;
+
+		ErrorDetails.TransactionMaxSizeExceeded transaction_max_size_exceeded = 5;
+	}
+}
+
+// AppendStreamOutput represents the output of appending records to a specific stream.
+message AppendStreamResponse {
+	// The result of the append operation.
+	oneof result {
+		// Success represents the successful outcome of an append operation.
+		AppendStreamSuccess success = 1;
+
+		// Failure represents the details of a failed append operation.
+		AppendStreamFailure failure = 2;
+	}
+}
+
+// MultiStreamAppendRequest represents a request to append records to multiple streams.
+message MultiStreamAppendRequest {
+	// A list of AppendStreamInput messages, each representing a stream to which records should be appended.
+	repeated AppendStreamRequest input = 1;
+}
+
+// Response from the MultiStreamAppend operation.
+message MultiStreamAppendResponse {
+	oneof result {
+		// Success represents the successful outcome of a multi-stream append operation.
+		Success success = 1;
+
+		// Failure represents the details of a failed multi-stream append operation.
+		Failure failure = 2;
+	}
+
+	message Success {
+		repeated AppendStreamSuccess output = 1;
+	}
+
+	message Failure {
+		repeated AppendStreamFailure output = 1;
+	}
+}
+
+// ErrorDetails provides detailed information about specific error conditions.
+message ErrorDetails {
+	// When the user does not have sufficient permissions to perform the operation.
+	message AccessDenied {
+		// The reason for access denial.
+		string reason = 1;
+	}
+
+	// When the stream has been deleted.
+	message StreamDeleted {
+	}
+
+	// When the expected revision of the stream does not match the actual revision.
+	message WrongExpectedRevision {
+		// The actual revision of the stream.
+		int64 stream_revision = 1;
+	}
+
+	// When the transaction exceeds the maximum size allowed
+	// (its bigger than the configured chunk size).
+	message TransactionMaxSizeExceeded {
+		// The maximum allowed size of the transaction.
+		int32 max_size = 1;
+	}
+}
+
+// todo: support for more types
+message DynamicValue {
```

Comment:
The clients actually must be able to send `string` values.

I would add our latest iteration of the DynamicValue right now but we don't really need to support all those types atm.

Reply:
:+1: i've added the other 'simple' types